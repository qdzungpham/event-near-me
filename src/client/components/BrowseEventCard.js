import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Hidden from '@material-ui/core/Hidden';
import Typography from '@material-ui/core/Typography';

const styles = theme => ({
  card: {
    display: 'flex',
    marginTop: theme.spacing.unit
  },
  cardHover: {
    display: 'flex',
    marginTop: theme.spacing.unit,
    'background-color': '#dddddd'
  },
  cardDetails: {
    flex: 1
  },
  cardMedia: {
    width: 160
  }
});

const formatDate = (dateString) => {
  const date = new Date(dateString);
  return date.toLocaleDateString('en-US', {
    weekday: 'short',
    month: 'short',
    day: 'numeric',
    hour12: true,
    hour: '2-digit',
    minute: '2-digit'
  });
};

const BrowseEventCard = (props) => {
  const {
    classes, event, overMarker, onHoverEventCard
  } = props;

  const onMouseEnter = () => {
    onHoverEventCard(event.id);
  };

  return (
    <Card
      onMouseEnter={onMouseEnter}
      className={overMarker !== event.id ? classes.card : classes.cardHover}
    >
      <Hidden xsDown>
        <CardMedia className={classes.cardMedia} image={event.image} title={event.title} />
      </Hidden>
      <div className={classes.cardDetails}>
        <CardContent>
          <Typography variant="headline">{event.title}</Typography>
          <Typography variant="subheading" color="textSecondary">
            {formatDate(event.dateTime)}
          </Typography>
          <Typography variant="subheading" color="textSecondary">
            {event.location}
          </Typography>
        </CardContent>
      </div>
    </Card>
  );
};

BrowseEventCard.propTypes = {
  classes: PropTypes.object.isRequired,
  event: PropTypes.object.isRequired,
  overMarker: PropTypes.string.isRequired,
  onHoverEventCard: PropTypes.func.isRequired
};

export default withStyles(styles)(BrowseEventCard);
