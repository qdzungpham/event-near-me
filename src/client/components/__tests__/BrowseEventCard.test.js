import React from 'react';
import { shallow } from 'enzyme';
import BrowseEventCard from '../BrowseEventCard';

describe('BrowseEventCard', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <BrowseEventCard
        event={{
          id: '2',
          title: 'title',
          dateTime: '2018-09-07T14:12:15.112Z',
          location: 'Brisbane',
          coords: [-24, 153],
          distance: 21,
          image: 'http://via.placeholder.com/100x100'
        }}
        overMarker="-1"
        onHoverEventCard={jest.fn}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
