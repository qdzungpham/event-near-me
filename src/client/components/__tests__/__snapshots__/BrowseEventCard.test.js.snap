// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowseEventCard renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(BrowseEventCard)
    event={
      Object {
        "coords": Array [
          -24,
          153,
        ],
        "dateTime": "2018-09-07T14:12:15.112Z",
        "distance": 21,
        "id": "2",
        "image": "http://via.placeholder.com/100x100",
        "location": "Brisbane",
        "title": "title",
      }
    }
    onHoverEventCard={[Function]}
    overMarker="-1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "card": "BrowseEventCard-card-1",
        "cardDetails": "BrowseEventCard-cardDetails-3",
        "cardHover": "BrowseEventCard-cardHover-2",
        "cardMedia": "BrowseEventCard-cardMedia-4",
      },
      "event": Object {
        "coords": Array [
          -24,
          153,
        ],
        "dateTime": "2018-09-07T14:12:15.112Z",
        "distance": 21,
        "id": "2",
        "image": "http://via.placeholder.com/100x100",
        "location": "Brisbane",
        "title": "title",
      },
      "onHoverEventCard": [Function],
      "overMarker": "-1",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "card": "BrowseEventCard-card-1",
          "cardDetails": "BrowseEventCard-cardDetails-3",
          "cardHover": "BrowseEventCard-cardHover-2",
          "cardMedia": "BrowseEventCard-cardMedia-4",
        },
        "event": Object {
          "coords": Array [
            -24,
            153,
          ],
          "dateTime": "2018-09-07T14:12:15.112Z",
          "distance": 21,
          "id": "2",
          "image": "http://via.placeholder.com/100x100",
          "location": "Brisbane",
          "title": "title",
        },
        "onHoverEventCard": [Function],
        "overMarker": "-1",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
